File: backend/src/main/java/com/festago/zfestival/dto/FestivalsResponse.java
package com.festago.zfestival.dto;

import static java.util.stream.Collectors.collectingAndThen;
import static java.util.stream.Collectors.toList;

import com.festago.zfestival.domain.Festival;
import java.util.List;

public record FestivalsResponse(
    List<FestivalResponse> festivals) {

    public static FestivalsResponse from(List<Festival> festivals) {
        return festivals.stream()
            .map(FestivalResponse::from)
            .collect(collectingAndThen(toList(), FestivalsResponse::new));
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/FestivalsResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.Festival
import kotlinx.serialization.Serializable

@Serializable
data class FestivalsResponse(
    val festivals: List<FestivalResponse>,
) {
    fun toDomain(): List<Festival> = festivals.map { it.toDomain() }
}
